aiohttp/multipart.py

```diff
@@ -633,11 +633,21 @@ class MultipartWriter(Payload):
 
     def __init__(self, subtype='mixed', boundary=None):
         boundary = boundary if boundary is not None else uuid.uuid4().hex
+        # The underlying Payload API demands a str (utf-8), not bytes,
+        # so we need to ensure we don't lose anything during conversion.
+        # As a result, require the boundary to be ASCII only.
+        # In both situations.
         try:
-            self._boundary = boundary.encode('us-ascii')
-        except UnicodeEncodeError:
-            raise ValueError('boundary should contains ASCII only chars')
-        ctype = 'multipart/{}; boundary="{}"'.format(subtype, boundary)
+            if isinstance(boundary, bytes):
```

Comment:
What is the case for binary boundary input?

Reply:
It is true to how it is actually used and defined in the spec. In this particular situations, users don't get more value out of it because we have to convert the boundary back to a unicode string for the Payload header. Otherwise characters above 0x80 could be used Insel the boundary. 

Ideally, all headers would work on byte strings (only or additionally), but I didn't consider this to be the scope of this pr.

Reply:
*bytes above 0x80 could be used. HTTP is a ascii-readable byte-based protocol.